server/src/main/java/org/elasticsearch/index/mapper/ContentPath.java

```diff
@@ -25,9 +28,21 @@ public ContentPath() {
         this.sb = new StringBuilder();
     }
 
-    String[] getPath() {
-        // used for testing
-        return path;
+    public String[] getPath() {
+        return path.clone();
+    }
+
+    public void setPath(String[] newPath) {
+        Objects.requireNonNull(newPath);
+        Arrays.fill(path, null);
+        index = 0;
+
+        for (String newPathPart : newPath) {
+            if (newPathPart == null) {
+                break;
+            }
+            add(newPathPart);
+        }
```

Comment:
Why not using `remove` and `add`? This new method is not needed.

Reply:
The only thing to remove here is the `_inference_fields` level no? It's maybe a bit more complicated on the add side since the semantic text can be under an object field but I don't understand the need to return what's removed.

Reply:
That approach would require assuming that the path when parsing the metafield is always `_inference_fields`, and hard-coding that assumption into the parsing logic. I'd like to implement a solution that does not involve an assumed path.

Reply:
Sorry but this doesn't make sense, we use a metadata field specifically to ensure that the path is always the same.